File: C++/average-salary-excluding-the-minimum-and-maximum-salary.cpp
class Solution {
public:
    double average(vector<int>& salary) {
        int low = salary[0];
        int high = salary[0];
        int sum = 0;
        int n = 0;
        for (int s: salary) {
            low = min(low, s);
            high = max(high, s);
            sum += s;
            n++;
        }
        sum = sum - high - low;
        return (double)sum / (n - 2);
    }
};

File: Kotlin/average-salary-excluding-the-minimum-and-maximum-salary.kt
class Solution {
    fun average(salary: IntArray): Double {
        var lo = salary[0]
        var hi = salary[0]
        var sum = 0.0
        for (n in salary) {
            sum += n
            if (lo > n)
                lo = n
            if (hi < n)
                hi = n
        }
        return (sum - lo - hi) / (salary.size - 2)
    }
}